src/turing.jl

```diff
@@ -0,0 +1,83 @@
+function from_turing(
+    chns=nothing;
+    model=nothing,
+    rng=Random.default_rng(),
+    nchains=ndraws = chns isa Turing.MCMCChains.Chains ? last(size(chns)) : 1,
+    ndraws=chns isa Turing.MCMCChains.Chains ? first(size(chns)) : 1_000,
+    library=Turing,
+    observed_data=nothing,
+    constant_data=nothing,
+    posterior_predictive=nothing,
+    prior=nothing,
+    prior_predictive=nothing,
+    log_likelihood=nothing,
+    kwargs...,
+)
+    groups = Dict{Symbol,Any}(
+        :observed_data => observed_data,
+        :constant_data => constant_data,
+        :posterior_predictive => posterior_predictive,
+        :prior => prior,
+        :prior_predictive => prior_predictive,
+        :log_likelihood => log_likelihood,
+    )
+    model === nothing && return from_mcmcchains(chns; library=library, groups..., kwargs...)
+    if groups[:prior] === nothing
+        groups[:prior] = reduce(
+            Turing.chainscat,
+            map(
+                _ -> Turing.sample(rng, model, Turing.Prior(), ndraws; progress=false),
+                1:nchains,
```

Comment:
Might want to just use the threaded version, which in the case of `JULIA_NUM_THREADS=1` just does this entire thing but the code is much shorter.